src/ralph/accounts/views.py

```diff
@@ -28,6 +31,51 @@ def user_licence(self, item):
             return []
 
 
+class InventoryTagConfirmationView(RalphBaseTemplateView):
+    template_name = 'ralphuser/inventory_tag_confirmation.html'
+
+    def get_context_data(self, **kwargs):
+        print(self.request.method)
+        context = super().get_context_data(**kwargs)
+        asset_fields = [
+            ('barcode', _('Barcode / Inventory Number')),
+            'model__category__name', 'model__manufacturer__name',
+            'model__name', ('sn', _('Serial Number'))
+        ]
+
+        context['asset_details'] = MyEquipmentAssetList(
+            BackOfficeAsset.objects.filter(id=self.kwargs['asset_id']),
+            asset_fields,
+            request=self.request
+        )
+        return context
+
+
+class InventoryTagView(RalphBaseTemplateView):
+    template_name = 'ralphuser/inventory_tag.html'
+    
+    def post(self, request, *args, **kwargs):
+        if not 'asset_id' in request.POST:
+            return HttpResponseBadRequest()
+
+        asset = BackOfficeAsset.objects.get(pk=request.POST['asset_id'])
+        if not asset.user.username == request.user.username:
+            return HttpResponseForbidden()
+
+        asset.tags.add(settings.INVENTORY_TAG)
+        asset.save()
+
+        return render(
+            request, 
+            self.template_name, 
+            context=self.get_context_data()
+        )
+
+    def get_context_data(self, **kwargs):
```

Comment:
Are you missing add some extra arguments to context? Or this block is not necessary?